exercise/static/exercise/results_staff.js

```diff
@@ -705,23 +705,39 @@
         }
     }
 
+    function gatherFromAPIPaging(url) {
+        function gather(cur_result, url) {
+            return $.ajax(url)
+            .then(function(response) {
+                cur_result = cur_result.concat(response.results)
+                if (response.next) {
+                    return gather(cur_result, response.next);
+                } else {
+                    return cur_result
+                }
+            }, function(reason) {
+                throw new Error("Pagination ajax failed " + reason);
+            });
+        }
+        return gather([], url);
+    }
 
     /*
      * The following code is responsible for handling all the ajax calls to get the data from /api/v2/
      * The code also initializes the module and exercise selection options.
      * Creates the table for the first time, when all ajax calls have finished.
      * READER WARNING: You are entering callback hell.
      */
-    $.when($.ajax(exercisesUrl), $.ajax(studentsUrl), $.ajax(usertagsUrl))
-        .done(function(exercisesResolve, studentsResolve, usertagsResolve) {
-        _exercises = exercisesResolve[0];
-        _students = studentsResolve[0];
-        _usertags = usertagsResolve[0];
-
-        let requiredPointAjaxCalls = _students.results.length;
-        let requiredUserAjaxCalls = _students.results.length;
+    $.when($.ajax(exercisesUrl), gatherFromAPIPaging(studentsUrl), $.ajax(usertagsUrl))
+        .done(function(exercisesResolve, studentsPagedResults, usertagsResolve) {
+        _exercises = exercisesResolve[0].results;
+        _students = studentsPagedResults;
+        _usertags = usertagsResolve[0].results;
```

Comment:
`_usertags = usertagsResolve[0].results;`
Does this mean that only the first page is used for the user tags and if there are over 100 tags, it would break?

Reply:
Thus, should both `exercisesUrl` and `usertagsUrl` be wrapped in `gatherFromAPIPaging`?

Reply:
Even though no course uses over 100 tags, it would be safer to just deal with the potential pagination.